### PR TITLE
add MagicNumber#ignoreNamedParameters option

### LINKED ISSUE #415: Update "MagicNumber" to not mark "Named Arguments" as code smells
Consider following example:

```kotlin
class HtlExpressionInsertHandler :
    HtlTextInsertHandler("=\"\${}\"", offset = 4)
```

`offset = 4` will be marked as "MagicNumber" rule violation.

The above technique is called [Named Arguments](https://kotlinlang.org/docs/reference/functions.html#named-arguments).
As you could see from the official documentation, the purpose of "Named Arguments" is:

* Improve convenience on caller side in case if called function has many parameters and/or parameters with default values
* Improve readability of code as argument names should explain the purpose of passed arguments (mostly for literals)

I think that current implementation of "MagicNumber" rule contradicts the second point, so it you should be updated as following:

"MagicNumber" passed via *Named Argument* should **not** be considered as code smell as the name of argument should explain the purpose of the number.

Thanks, 
Dmytro

### PR DESCRIPTION
**Pull request by Mauin**
Resolves #415

As @schalkms suggested I set this to true by default in the default config and Readme.md

### CODE REVIEW

**Comment by Mauin** [comment]
Oh, didn't realize that #426 also resolves #415. I'll leave this one open for now, but feel free to close this one in favor of the older PR if you want 🙃

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumberSpec.kt [hunk 1]
```diff
@@ -297,6 +297,34 @@ class MagicNumberSpec : Spek({
...
+		it("should be reported") {
+			val findings = MagicNumber().lint(code)
+			assertThat(findings).hasSize(1)
+		}
+
+		it("should not be reported when named parameters are ignored") {
+			val findings = MagicNumber(TestConfig(mapOf(MagicNumber.IGNORE_NAMED_PARAMETERS to "true"))).lint(code)
+			assertThat(findings).isEmpty()
+		}
+	}
```
[the remark this answers is not in the document]
Comment (Mauin) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/MagicNumberSpec.kt]: You're right! I've added some more checks and tests to verify that expressions like your example are still reported as magic numbers even if they are passed as named parameters.

There is already a test for the second case. It's verifying that the (fake) method call `val number = calculateNumber(four = 4)` behaves correctly.

**Comment by arturbosch** [comment]
@Mauin @DmytroTroynikov which PR now to merge oO. On the old one I was just waiting for the changes @schalkms requested to be added hmmm ... ? :)

**Comment by Mauin** [comment]
@arturbosch Your call. I simply did not realize #426 was already there when I created this PR. I'll close this one. Feel free to reopen and merge if needed, but I did not intend to steal @DmytroTroynikov's first-time contribution 👍

**Pull request closed without merging**